certora/spec/CVL2/timelockAuthorizerTed.spec

```diff
@@ -0,0 +1,556 @@
+import "timelockAuthorizerMain.spec";
+
+
+invariant hasPermissionIfIsGrantedOnTarget(bytes32 id, address account, address where)
+    isPermissionGrantedOnTarget(id, account, where) => hasPermission(id, account, where)
+
+
+// STATUS - verified
+// https://prover.certora.com/output/40577/43649cf8b8f148a98a7fb83a7647cc34/?anonymousKey=f4c1e5e509629a8ad342dc1997b0cc430264cd84
+rule rootNotZero(env e, method f)
+{
+    require(getRoot() != 0);
+    require(e.msg.sender != 0);
+
+    // Invoke any function
+    calldataarg args;
+    f(e, args);
+
+    assert getRoot() != 0;
+}
+
+
+// STATUS - verified
+// https://prover.certora.com/output/40577/e38e4eb130054ebb88922acd43f27777/?anonymousKey=d847f9e3d234739e3652f5e73b5d2dce9d404039
+rule onlyOneRoleChangeAtATimeForNonClaimRoot(env e, method f)
```

Comment:
- could you add some comments? becuase it's confusing a little bit or at least add meaning of the property
- maybe even better to split it into 3 smaller rules for each status: cancel, revoke, grant.

Reply:
Thank you for reminding me to add comments. I added the following one:
"When execution function other than the claimRoot, only one of the roles revoker, granter
  or canceler can change during the function call and this can happen for one entity only."
Regarding splitting the rule: The 3 rules would not be significantly smaller, as you would still need almost all the variables you have in the original rule. We would have to keep everything in these rules except for some asserts.